pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -47,10 +68,17 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		machine:       machine,
 	})
 	if err != nil {
-		return fmt.Errorf(scopeFailFmt, machine.Name, err)
+		fmtErr := fmt.Errorf(scopeFailFmt, machine.Name, err)
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration("error decoding MachineProviderConfig: %v", fmtErr), createEventAction)
 	}
 	defer scope.Close()
-	return newReconciler(scope).create()
+	if err := newReconciler(scope).create(); err != nil {
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration("error creating Machine: %v", err), createEventAction)
+	}
+	// This event is best-effort and might get missed in case of timeout
+	// on waitUntilOperationCompleted
+	a.eventRecorder.Eventf(machine, corev1.EventTypeNormal, "Created", "Created Machine %v", machine.Name)
```

Comment:
if we create constants why don't use them https://github.com/openshift/cluster-api-provider-gcp/pull/17/files#diff-b314f951e6142d0ce7df84e1af378d85R22